File: app/src/main/java/com/example/mesplats/IngredientDAO.java
package com.example.mesplats;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
interface IngredientDAO {

    @Query("SELECT * FROM ingredient")
    fun getAll(): List<Ingredient>

//    @Insert
//    fun insert(vararg ingredient: Ingredient)
//
//    @Delete
//    fun delete(ingredient: Ingredient)
//
//    @Query("SELECT * FROM Ingredient WHERE name LIKE :queryName")
//    fun findByName(queryName: String): Ingredient
}

File: app/src/main/java/com/example/mesplats/IngredientDao.kt
package com.example.mesplats

import androidx.room.*

@Dao
interface IngredientDao {

    @Query("SELECT * FROM ingredient ORDER BY name")
    fun getAll(): List<Ingredient>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(vararg ingredient: Ingredient)

    @Query("DELETE FROM ingredient")
    fun deleteAll()

    @Delete
    fun delete(ingredient: Ingredient)

    @Query("SELECT * FROM ingredient WHERE name LIKE :queryName")
    fun findByName(queryName: String): List<Ingredient>

    @Query("SELECT * FROM ingredient WHERE name LIKE :queryId")
    fun findById(queryId: Int): Ingredient
}